### ISSUE TITLE
Support minimal, narrow display mode

### ISSUE THREAD

**Issue by quinncomendant** [notable/notable#421]
Sometimes, it is useful to have the notes editor off to the side of the screen in a very narrow orientation. For example, when watching a video and taking notes, or quickly copy and pasting content from a browser window that is filling most of the screen. Currently, Notable's window can't be very narrow (minimum width appears to be around 700-800px) and to maximize the editor's window area it requires entering focus mode, which reduces available functionality (e.g., notes list is unavailable, Find (`⌘F`) doesn't work in focus mode).

To make this work, there will need to be some changes:

- Allow narrower window widths (down to 300px wide would be ideal).
- Auto-enter an pseudo focus mode at narrow window widths (hide all unnecessary UI elements).
- Option to view the list of notes above the note content area.
- Navigating notes easily: enable find and navigating between notes with the keyboard.
- Option to hide toolbar.

For Notable to work for me, I would need to be able to use Notable like this. I'm migrating from nvAlt, and my window orientation [looks like this](https://i.imgur.com/g5ePQzL.png) 90% of the time.

**Comment by fabiospampinato** [comment]
> * Allow narrower window widths (down to 300px wide would be ideal).

The problem is that if the width of the window is lower than what's currently allowed there might be not enough space for displaying everything that we currently display.

> * Auto-enter an pseudo focus mode at narrow window widths (hide all unnecessary UI elements).

I think this would be a bit confusing. You can expect this sorts of behavior from a responsive website, but usually I don't think native apps behave like that. 

Maybe we could allow for smaller windows when in focus mode though.

> * Option to view the list of notes above the note content area.

This seems pretty specific, I don't think this is going to happen, but maybe you could do that yourself with some custom CSS in the future.

> * Navigating notes easily: enable find and navigating between notes with the keyboard.

Right now you can already navigate tags and notes via shortcuts, but only "blindly", once we implement a quick-open-like component #338 that I think this issue should be mostly resolved.

> * Option to hide toolbar.

Basically a "Presentation Mode" 🤔

**Comment by lmihalkovic** [comment]
yes... I was thinking about that too ... once you find the right doc, it is nice to be able to just focus on its contents. Most editors call it some variations of 'zen mode'

**Comment by quinncomendant** [comment]
> once you find the right doc, it is nice to be able to just focus on its contents. 

Exactly! I'd say only 3% of the time are you actually searching for or navigating between notes, and 97% of the time you are in a note's content and don't want to see any of the note-selection UI elements. 

I understand it's a pretty radical change from the current UI to support a no-frills editor-only mode *which also has access to a note-navigation tools*. But that's something that's important to me.

**Comment by CavalloneChen** [comment]
This would be a very helpful feature.

**Comment by fabiospampinato** [comment]
A small update on this:

- The "no distraction" part of this request has been implemented in the form of what we call "zen mode".

- The ability to switch between notes in that situation has been fixed with the addition of the quick open panel.

- The only part of this request left to implement is actually allowing the app to be resized to narrower widths. I've actually created a narrow note-taking app in the past called [Noty](https://github.com/fabiospampinato/noty#demo) and I see no point in maintaining 2 note-taking apps, so allowing a narrow display mode that allows me to replace Noty with Notable would also be pretty useful to me. I kind of have a design in mind of how this should work, but in order to do it properly it will take some time. This would also be important for allowing the PWA #83 to work properly on mobile.

**Comment by nkershaw** [comment]
To tack on an idea here, it would be great to have the note editor center content to the middle of the editor window in Zen mode, similar to what [Goyo](https://github.com/junegunn/goyo.vim) and other writers (e.g. [GhostWriter](https://thenextweb.com/apps/2018/01/04/ghostwriter-gorgeous-distraction-free-markdown-editor-windows-linux/)) do:

![image](https://user-images.githubusercontent.com/38503871/66840910-cb530980-ef36-11e9-9b25-9b196ed0c94f.png)

**Comment by fabiospampinato** [comment]
> The only part of this request left to implement is actually allowing the app to be resized to narrower widths.

And that's been implemented too, so I'm closing this.

@nkershaw Please open a separate issue for that.
